Apps/W1/EDocumentConnectors/FORNAV/app/src/ForNAV/IncomingEDocsApi.Page.al

```diff
@@ -0,0 +1,145 @@
+                field(doc; Document)
+                {
+                    ApplicationArea = All;
+                }
+                field(message; Message)
+                {
+                    ApplicationArea = All;
+                }
+                field(status; Rec.Status)
+                {
+                    ApplicationArea = All;
+                }
+                field(eDocumentType; Rec.EDocumentType)
+                {
+                    ApplicationArea = All;
+                }
+                field(schemeID; Rec.SchemeID)
+                {
+                    ApplicationArea = All;
+                }
+                field(endpointID; Rec.EndpointID)
+                {
+                    ApplicationArea = All;
+                }
+            }
+        }
+    }
+
+    var
+        Document, Message : BigText;
+
+    [TryFunction]
+    local procedure ScheduleJob(JobQueueCodeunit: Integer; RecId: RecordId)
+    var
+        QueueEntry: Record "Job Queue Entry";
+        // Enqueue: Codeunit "Job Queue - Enqueue";
+        PeppolJobQueue: Codeunit "ForNAV Peppol Job Queue";
+    begin
+        QueueEntry.ID := CreateGuid();
+        QueueEntry."Record ID to Process" := RecId;
+        QueueEntry."Object ID to Run" := JobQueueCodeunit;
+        QueueEntry."Object Type to Run" := QueueEntry."Object Type to Run"::Codeunit;
+        QueueEntry."Job Queue Category Code" := PeppolJobQueue.GetForNAVCategoryCode();
+        QueueEntry.Description := 'Used by ForNAV to process incoming e-documents';
+        QueueEntry.Status := QueueEntry.Status::"On Hold";
+        QueueEntry.Insert();
+        // Enqueue.Run(QueueEntry);
+    end;
+
+    local procedure SetErrorMessage(var NewMessage: BigText)
+    var
+        Error: ErrorInfo;
+    begin
+        Clear(NewMessage);
+        NewMessage.AddText('Error\n');
+        NewMessage.AddText(GetLastErrorText() + '\n');
+        NewMessage.AddText(GetLastErrorCallStack() + '\n');
+        foreach Error in GetCollectedErrors() do
+            NewMessage.AddText(Error.Message + '\n');
+    end;
+
+    trigger OnInsertRecord(BelowxRec: Boolean): Boolean
+    var
+        EDocumentService: Record "E-Document Service";
+        BlankRecordId: RecordId;
+        DocOs, MessageOs : OutStream;
+    begin
+        case Rec.DocType of
+            Rec.DocType::Evidence:
+                if not ScheduleJob(Codeunit::"E-Document Get Response", BlankRecordId) then
+                    SetErrorMessage(Message);
+            Rec.DocType::ApplicationResponse:
+                if not ScheduleJob(Codeunit::"ForNAV App. Resp. Handler", Rec.RecordId()) then
+                    SetErrorMessage(Message);
+            Rec.DocType::Invoice, Rec.DocType::CreditNote:
+                begin
+                    if not EDocumentService.Get('FORNAV') then
+                        exit(false);
+
+                    if not ScheduleJob(6147, EDocumentService.RecordId()) then // Codeunit::"E-Document Import Job"
```

Comment:
Cant you reference the import job. Something wrong in the design here, if you need to rely on the const id.